### ISSUE TITLE
[DatePicker] Year overflow broken

### ISSUE THREAD

**Issue by DylanSimowitz** [mui/mui-x#7495]
### Problem description

I noticed in the docs that the datepicker should go 100 years backwards and 100 years forwards by default, however in the docs and in any datepicker I make it only goes to 2013. I found the issue to be justify-content: center. This centers the flex item vertically because the flex-direction is column. For some reason overflow doesn't take into account the height before the center point of the flex item. Setting the flex-direction to row (default value) fixes this issue and I can see all the way to 1916. Unfortunately this breaks positioning of datepicker ranges that do not overflow. They are no longer vertically centered. Setting align-items: center to center non-overflowed ranges will subsequently break our original fix of overflowed ranges for the same reason that justify-content breaks it in the current implementation.

TL;DR: Flexbox centering and overflow don't work nicely.
### Steps to reproduce

Look at the Simple example in the docs.
### Versions
- Material-UI: 15
- React: 15
- Browser: Chrome/Firefox

### Images & references

![](https://i.imgur.com/FSI2jDr.png)
